plugins/CoreAdminHome/Commands/InvalidateReportData.php

```diff
@@ -0,0 +1,272 @@
+            }
+        }
+    }
+
+    private function getSitesToInvalidateFor(InputInterface $input)
+    {
+        $sites = $input->getOption('sites');
+
+        $siteIds = Site::getIdSitesFromIdSitesString($sites);
+        if (empty($siteIds)) {
+            throw new \InvalidArgumentException("Invalid --sites value: '$sites'.");
+        }
+
+        $allSiteIds = SitesManagerAPI::getInstance()->getAllSitesId();
+        foreach ($siteIds as $idSite) {
+            if (!in_array($idSite, $allSiteIds)) {
+                throw new \InvalidArgumentException("Invalid --sites value: '$sites', there are no sites with IDs = $idSite");
+            }
+        }
+
+        return $siteIds;
+    }
+
+    private function getPeriodTypesToInvalidateFor(InputInterface $input)
+    {
+        $periods = $input->getOption('periods');
+        if (empty($periods)) {
+            throw new \InvalidArgumentException("The --periods argument is required.");
+        }
+
+        if ($periods == self::ALL_OPTION_VALUE) {
+            $result = array_keys(Piwik::$idPeriods);
+            unset($result[4]); // remove 'range' period
+            return $result;
+        }
+
+        $periods = explode(',', $periods);
+        $periods = array_map('trim', $periods);
+
+        foreach ($periods as $periodIdentifier) {
+            if ($periodIdentifier == 'range') {
+                throw new \InvalidArgumentException(
+                    "Invalid period type: invalidating range periods is not currently supported.");
+            }
+
+            if (!isset(Piwik::$idPeriods[$periodIdentifier])) {
+                throw new \InvalidArgumentException("Invalid period type '$periodIdentifier' supplied in --periods.");
+            }
+        }
+
+        return $periods;
+    }
+
+    /**
+     * @param InputInterface $input
+     * @return Date[][]
+     */
+    private function getDateRangesToInvalidateFor(InputInterface $input)
+    {
+        $dateRanges = $input->getOption('dates');
+        if (empty($dateRanges)) {
+            throw new \InvalidArgumentException("The --dates option is required.");
+        }
+
+        $result = array();
+        foreach ($dateRanges as $dateRangeString) {
+            $parts = explode(',', $dateRangeString);
+
+            if (count($parts) == 1) {
+                $parts[1] = $parts[0];
+            } else if (count($parts) > 2) {
+                throw new \InvalidArgumentException("Invalid date range specifier: '$dateRangeString'.");
+            }
+
+            // validate dates
+            $startDate = $this->validateDate($parts[0]);
+            $endDate = $this->validateDate($parts[1]);
+
+            $result[] = array($startDate, $endDate);
+        }
+        return $result;
+    }
+
+    /**
+     * @param string[] $periodTypes
+     * @param Date[][] $dateRanges
+     * @param bool $cascade
+     * @return Date[][]
+     */
+    public function getPeriodsToInvalidateFor($periodTypes, $dateRanges, $cascade)
```

Comment:
This one is public although not needed to have it public here right? I think it's public so that it can be tested so maybe it can be moved in another class as well where it is public and maybe "needed" to be public